src/FSharpLint.Core/Rules/Conventions/Naming/NamingHelper.fs

```diff
@@ -361,3 +361,14 @@ let rec identFromSimplePat = function
     | SynSimplePat.Id(ident, _, _, _, _, _) -> Some(ident)
     | SynSimplePat.Typed(p, _, _) -> identFromSimplePat p
     | SynSimplePat.Attrib(_) -> None
+
+let rec isNested args nodeIndex =
+    let parent = args.SyntaxArray.[nodeIndex].ParentIndex
+    let actual = args.SyntaxArray.[parent].Actual
+
+    match actual with
+    | AstNode.ModuleDeclaration (SynModuleDecl.Types (_, _)) -> false
+    | AstNode.ModuleDeclaration (SynModuleDecl.NestedModule (_, _, _, _, _)) -> false
+    | AstNode.Binding (SynBinding (_, _, _, _, _, _, _, _, _, _, _, _)) -> false
+    | AstNode.Expression (SynExpr.LetOrUse (_, _, _, _, _)) -> true
```

Comment:
this match block can be hugely simplified if you move the case `AstNode.Expression (SynExpr.LetOrUse (_, _, _, _, _))` to the first position, and then just use `_ -> false` as a second case